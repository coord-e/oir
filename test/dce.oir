# test: "$OIR_BIN" -p liveness,dce
LABEL entry
  v1 = IMM 10  # dead
  v2 = IMM 20  # dead
  v2 = IMM 30
  JUMP loop
LABEL loop
  v3 = IMM 10
  v4 = EQUAL v2 v3
  v2 = IMM 10  # not dead
  BRANCH v4 end loop
LABEL end
  v5 = IMM 20
  RETURN v5
