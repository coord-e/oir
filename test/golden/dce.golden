# OIR entry=0, exit=2
# BB0 succs={1} preds={}
LABEL entry  # 0
  v2 = IMM 30
  JUMP loop  # 1
# BB1 succs={2, 1} preds={0, 1}
LABEL loop  # 1
  v3 = IMM 10
  v4 = EQUAL v2 v3
  v2 = IMM 10
  BRANCH v4 end loop  # 2, 1
# BB2 succs={} preds={1}
LABEL end  # 2
  v5 = IMM 20
  RETURN v5
